doc/dev/cephadm/compliance-check.rst

```diff
@@ -0,0 +1,119 @@
+================
+Compliance Check
+================
+
+The stability and reliability of a Ceph cluster is not just dependent upon the Ceph daemons, but
+also the OS and hardware that Ceph is installed on. This document is intended to promote a design 
+discussion for providing a "compliance" feature within mgr/cephadm, which would be responsible for
+identifying common platform related issues that could impact Ceph stability and operation.
+
+The ultimate goal of the checks is to identify issues early and raise a healthcheck WARN
+event, to alert the Administrator to the issue.
+
+Prerequisites
+=============
+In order to effectively analyse the hosts that Ceph is deployed to, this feature requires a cache
+of host related metadata. The metadata is already available from cephadm's HostFacts class and the
+gather-facts cephadm command. For the purposes of this document, we will assume that this
+data is available within the mgr/cephadm "cache" structure.
+
+Some checks will require that the host status is also populated e.g. ONLINE, OFFLINE, MAINTENANCE
+
+Administrator Interaction
+=========================
+Not all users will require this feature, and must be able to 'opt out'. For this reason,
+mgr/cephadm must provide controls, such as the following;
+
+.. code-block::
+
+   ceph cephadm compliance enable | disable | status [--format json]
```

Comment:
@pcuzner : can you explain the difference between the "compliance ls" command and the "compliance status"?

Reply:
The doc already covers that just after the code block that defines the proposed command interactions. Status is highlevel, is it enabled/active, what's the check interval...whereas ls is the list of the checks and whether they are enabled or not.

Reply:
Understood. I think we should add some way to force the checks execution manually. Maybe a new command
ceph cephadm compliance run
What do you think?